src/queue.js

```diff
@@ -0,0 +1,55 @@
+'use strict'
+
+function Node( data ) {
+  this._data = data
+  this._next = null
+}
+
+export default class Queue {
+  constructor() {
+    this._front = null
+    this._back = null
+    this._length = 0
+  }
+
+  length() {
+    return this._length
+  }
+
+  isEmpty() {
+    return this._length == 0
+  }
+
+  back() {
+    return this._back._data
+  }
+
+  front() {
+    return this._front._data
+  }
+
+  enqueue( data ) {
+    let newNode = new Node( data )
+    let oldBack = this._back
+    if ( this._length === 0 ) {
+      this._front = newNode
+      this._back = newNode
+    } else {
+      this._back = newNode
```

Comment:
This line is in both the `if` and `else` paths in the conditional.

Place it outside the conditional so it runs in both cases and you don't repeat yourself (DRY principle).